tests/test_generator/test_lengths.py

```diff
@@ -0,0 +1,59 @@
+import unittest
+
+from gutenTAG import GutenTAG
+
+
+class TestLengths(unittest.TestCase):
+    def setUp(self) -> None:
+        self.config = {
+            "timeseries": [
+                {
+                    "name": "test-amp",
+                    "length": 500,
+                    "base-oscillations": [{"kind": "sine"}],
+                    "anomalies": [
+                        {
+                            "position": "end",
+                            "length": 47,
+                            # "creeping-length": 12,  # remove to test the default 20% option
+                            "channel": 0,
+                            "kinds": [{"kind": "amplitude", "amplitude_factor": 2.0}],
+                        }
+                    ],
+                }
+            ]
+        }
+        self.creeping_factor = 0.3
+
+    def update_config(self, length: int, include_creeping: bool = False):
+        self.config["timeseries"][0]["anomalies"][0]["length"] = length
+        cl = int(round(length * self.creeping_factor)) if include_creeping else 0
+        self.config["timeseries"][0]["anomalies"][0]["creeping-length"] = cl
+
+    def __run(self):
```

Comment:
We use single `_` to mark private methods.

You can use assertions in functions called from the original test function. I would have all the assertions in this function.

```suggestion
    def _run_and_validate(self, config: Dict[str, Any]) -> None:
```